lib/TPP/DefaultTppPasses.cpp

```diff
@@ -141,8 +139,12 @@ struct DefaultTppPasses
             BrgemmLinalgTilingOptions{lhsTile, rhsTile}));
         pm.addNestedPass<func::FuncOp>(createLoopInvariantCodeMotionPass());
         pm.addNestedPass<func::FuncOp>(createVectorizationPass());
-        pm.addNestedPass<func::FuncOp>(createCanonicalizerPass());
-        if (vectorToXSMM) {
+
+	//Please note, canonicalizer should be after hoisting pass because 
+	//it fuses outer tiling loops and it results in no pattern 
+	//matching for hoisting pass. Moved inside VectorToKernel Path.
```

Comment:
@KavithaTipturMadhu, does `vectorToXSMM` need the canonicalization pass? The behaviour here will be changed for that path. We may need to add it to your bundle.

Reply:
Let's leave it at that, since the vector-to-xsmm work is not complete yet. Kavitha can fix this on her next PR, if needed.